src/code_105700_len_2130.c

```diff
@@ -1,99 +1,213 @@
 #include "common.h"
 
-INCLUDE_ASM(s32, "code_105700_len_2130", func_802E3E80);
-
-INCLUDE_ASM(s32, "code_105700_len_2130", func_802E3E9C);
+typedef struct struct802E3F0C {
+    /* 0x00 */ char unk_00[10];
+    /* 0x0A */ u16 unk_A;
+    /* 0x0C */ char unk_C[4];
+    /* 0x10 */ s16 unk_10;
+    /* 0x12 */ s16 unk_12;
+} struct802E3F0C;
+
+void func_802E3E80(Entity* entity) {
+    func_802E3650(entity);
+}
+
+void func_802E3E9C(Entity* entity) {
+    s32 alpha;
+
+    alpha = entity->alpha;
+    if ((gPlayerStatus.animFlags & 1)) {
+        alpha += 0x20;
+        if (alpha > 0xC0) {
+            alpha = 0xC0;
+        }
+        entity->alpha = alpha;
+        return;
+    } else {
+        alpha -= 0x20;
+        if (alpha <= 0) {
+            alpha = 0;
+        }
+        entity->alpha = alpha;
+        return;
+    }
+}
 
 INCLUDE_ASM(s32, "code_105700_len_2130", func_802E3EE0);
 
-INCLUDE_ASM(s32, "code_105700_len_2130", func_802E3F0C);
-
-INCLUDE_ASM(s32, "code_105700_len_2130", func_802E4040);
-
-INCLUDE_ASM(s32, "code_105700_len_2130", func_802E405C);
-
-INCLUDE_ASM(s32, "code_105700_len_2130", func_802E40A0);
-
-INCLUDE_ASM(s32, "code_105700_len_2130", func_802E4108);
-
-INCLUDE_ASM(s32, "code_105700_len_2130", func_802E4154);
-
-INCLUDE_ASM(s32, "code_105700_len_2130", func_802E421C);
-
-INCLUDE_ASM(s32, "code_105700_len_2130", func_802E4484);
-
-INCLUDE_ASM(s32, "code_105700_len_2130", func_802E44CC);
-
-INCLUDE_ASM(s32, "code_105700_len_2130", func_802E44F8);
-
-INCLUDE_ASM(s32, "code_105700_len_2130", func_802E4540);
-
+INCLUDE_ASM(void, "code_105700_len_2130", func_802E3F0C, Entity* entity);
+
+void func_802E4040(Entity* entity) {
+    func_80110678(entity);
+}
+
+void func_802E405C(Entity* entity) {
+    D_802EB3B0 = func_80072230(0, entity->position.x, entity->position.y, entity->position.z, 1.0f, 0x3C);
+}
+
+void func_802E40A0(Entity* entity) {
+    s32 entityType;
+    s32 virtualModelIndex;
+    UNK_PTR ptr;
+
+    entityType = get_entity_type(entity->listIndex);
+    if ((entityType == 0x12) || (entityType == 0x14)) {
```

Comment:
Do we have an enum for this? If so we should probably use it (relevant elsewhere too)

Reply:
does not appear so.